src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/HostingStartupScenario.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Diagnostics.Monitoring.HostingStartup;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System.CommandLine;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios
+{
+    internal sealed class HostingStartupScenario
+    {
+        public static CliCommand Command()
+        {
+            CliCommand aspnetAppNoHostingStartupCommand = new(TestAppScenarios.HostingStartup.SubScenarios.VerifyAspNetAppWithoutHostingStartup);
+            aspnetAppNoHostingStartupCommand.SetAction(VerifyAspNetAppWithoutHostingStartupAsync);
+
+            CliCommand aspnetAppCommand = new(TestAppScenarios.HostingStartup.SubScenarios.VerifyAspNetApp);
+            aspnetAppCommand.SetAction(VerifyAspNetAppAsync);
+
+            CliCommand nonAspNetAppCommand = new(TestAppScenarios.HostingStartup.SubScenarios.VerifyNonAspNetAppNotImpacted);
+            nonAspNetAppCommand.SetAction(VerifyNonAspNetAppNotImpactedAsync);
+
+            CliCommand scenarioCommand = new(TestAppScenarios.HostingStartup.Name);
+            scenarioCommand.Subcommands.Add(aspnetAppNoHostingStartupCommand);
+            scenarioCommand.Subcommands.Add(aspnetAppCommand);
+            scenarioCommand.Subcommands.Add(nonAspNetAppCommand);
+            return scenarioCommand;
+        }
+
+        public static Task<int> VerifyNonAspNetAppNotImpactedAsync(ParseResult result, CancellationToken token)
+        {
+            return ScenarioHelpers.RunScenarioAsync(logger =>
+            {
+                return Task.FromResult(0);
+            }, token);
+        }
+
+        public static Task<int> VerifyAspNetAppWithoutHostingStartupAsync(ParseResult result, CancellationToken token)
+        {
+            return ScenarioHelpers.RunWebScenarioAsync<Startup>((logger, host) =>
+            {
+                Assert.Null(host.Services.GetService<InProcMonitoringSentinelService>());
```

Comment:
Because you are referencing types directly from the hosting startup assembly, that likely means the hosting startup assembly is part of the application output and is resolved using the built in assembly resolver. I think it would be better if the test app did not reference the assembly directly and allowed the startup hook assembly to resolve it. That will be probably require removing the project reference and using reflection to access the necessary parts of the hosting startup from the test app.

This doesn't have to be done as part of this PR since this is already happening without this PR.

Reply:
/todo [Parameter Capturing] Remove hosting startup project dependency from unit test application